receivers/mqtt/config.go

```diff
@@ -16,14 +16,16 @@ const (
 )
 
 type Config struct {
-	BrokerURL          string `json:"brokerUrl,omitempty" yaml:"brokerUrl,omitempty"`
-	ClientID           string `json:"clientId,omitempty" yaml:"clientId,omitempty"`
-	Topic              string `json:"topic,omitempty" yaml:"topic,omitempty"`
-	Message            string `json:"message,omitempty" yaml:"message,omitempty"`
-	MessageFormat      string `json:"messageFormat,omitempty" yaml:"messageFormat,omitempty"`
-	Username           string `json:"username,omitempty" yaml:"username,omitempty"`
-	Password           string `json:"password,omitempty" yaml:"password,omitempty"`
-	InsecureSkipVerify bool   `json:"insecureSkipVerify,omitempty" yaml:"insecureSkipVerify,omitempty"`
+	BrokerURL     string                   `json:"brokerUrl,omitempty" yaml:"brokerUrl,omitempty"`
+	ClientID      string                   `json:"clientId,omitempty" yaml:"clientId,omitempty"`
+	Topic         string                   `json:"topic,omitempty" yaml:"topic,omitempty"`
+	Message       string                   `json:"message,omitempty" yaml:"message,omitempty"`
+	MessageFormat string                   `json:"messageFormat,omitempty" yaml:"messageFormat,omitempty"`
+	Username      string                   `json:"username,omitempty" yaml:"username,omitempty"`
+	Password      string                   `json:"password,omitempty" yaml:"password,omitempty"`
+	QoS           receivers.OptionalNumber `json:"qos,omitempty" yaml:"qos,omitempty"`
+	Retain        bool                     `json:"retain,omitempty" yaml:"retain,omitempty"`
+	TLS           *receivers.TLSConfig     `json:"tls,omitempty" yaml:"tls,omitempty"`
```

Comment:
Let's call it `TLSConfig` and use json name `tls_config`, the same as in https://github.com/grafana/prometheus-alertmanager/blob/3d49ff83c74e253fc60b7659d952b24c59c44962/config/notifiers.go#L264